### PR TITLE
Fix identifying initializers in ONNX graph simplification

### PR DESCRIPTION
**Pull request by fengyuentau**
Fixes https://github.com/opencv/opencv/issues/23295

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/onnx/onnx_graph_simplifier.cpp [hunk 1]
```diff
@@ -192,20 +186,18 @@ class GeluSubGraph : public Subgraph
     {
         if (Subgraph::match(net, nodeId, matchedNodesIds, targetNodesIds))
         {
-            bool withInitializer = isWithInitializer(matchedNodesIds);
-
             // Check Div[B=sqrt(2)]
-            float divisor = extractConstant(net, matchedNodesIds[0], 1, withInitializer);
+            float divisor = extractConstant(net, matchedNodesIds[0], 1);
             if (divisor - M_SQRT2 >= 1e-6)
```
Comment (vpisarev) [inline modules/dnn/src/onnx/onnx_graph_simplifier.cpp:191]: why don't we use fabs() here? it's not a part of your patch, but this place looks strange

File: modules/dnn/src/onnx/onnx_graph_simplifier.cpp [hunk 2]
```diff
@@ -192,20 +186,18 @@ class GeluSubGraph : public Subgraph
...
                 return false;
 
             // Check Add[B=1]
-            float add_const = extractConstant(net, matchedNodesIds[2], 1, withInitializer);
+            float add_const = extractConstant(net, matchedNodesIds[2], 1);
             if (add_const - 1.f >= 1e-6)
```
Comment (vpisarev) [inline modules/dnn/src/onnx/onnx_graph_simplifier.cpp:196]: same

File: modules/dnn/src/onnx/onnx_graph_simplifier.cpp [hunk 3]
```diff
@@ -192,20 +186,18 @@ class GeluSubGraph : public Subgraph
...
                 return false;
 
             // Check Mul[B=0.5]
-            float mul_const = extractConstant(net, matchedNodesIds[4], 1, withInitializer);
+            float mul_const = extractConstant(net, matchedNodesIds[4], 1);
             if (mul_const - 0.5f >= 1e-6)
```
Comment (vpisarev) [inline modules/dnn/src/onnx/onnx_graph_simplifier.cpp:201]: same

File: modules/dnn/src/onnx/onnx_graph_simplifier.cpp (hunk 1 above)
Comment (fengyuentau) [reply modules/dnn/src/onnx/onnx_graph_simplifier.cpp:191]: Sure. It is fixed with the use of fabs().

File: modules/dnn/src/onnx/onnx_graph_simplifier.cpp (hunk 2 above)
Comment (fengyuentau) [reply modules/dnn/src/onnx/onnx_graph_simplifier.cpp:196]: Fixed.

File: modules/dnn/src/onnx/onnx_graph_simplifier.cpp (hunk 3 above)
Comment (fengyuentau) [reply modules/dnn/src/onnx/onnx_graph_simplifier.cpp:201]: Fixed.

**Review by vpisarev — APPROVED** [review]

**Pull request merged**